UnityProject/Assets/Scripts/UI/Objects/Research/ResearchServer/GUI_ResearchServer.cs

```diff
@@ -0,0 +1,122 @@
+using UnityEngine;
+using UI.Core.NetUI;
+using System.Collections;
+using Systems.Research.Objects;
+using Systems.Research.Data;
+using Systems.Research;
+
+namespace UI.Objects.Research
+{
+	public class GUI_ResearchServer : NetTab
+	{
+		[SerializeField]
+		private EmptyItemList ResearchedTechList;
+
+		[SerializeField]
+		private EmptyItemList AvailableTechList;
+
+		[SerializeField]
+		private EmptyItemList FutureTechList;
+
+		[SerializeField]
+		private NetText_label PointLabel;
+
+		private Techweb techWeb;
+		private bool isUpdating = false;
+
+		public void Awake()
+		{
+			StartCoroutine(WaitForProvider());
+		}
+		
+		public void OnDestroy()
+		{
+			techWeb.UIupdate -= UpdateGUI;
+		}
+
+		private IEnumerator WaitForProvider()
+		{
+			while (Provider == null)
+			{
+				yield return WaitFor.EndOfFrame;
+			}
+
+			techWeb = Provider.GetComponent<ResearchServer>().Techweb;
+			techWeb.UIupdate += UpdateGUI;
+
+			if (!CustomNetworkManager.Instance._isServer) yield break;
+
+			UpdateGUI();
+
+			OnTabOpened.AddListener(UpdateGUIForPeepers);
+
+		}
+
+		public void UpdateGUIForPeepers(PlayerInfo notUsed)
+		{
+			if (!isUpdating)
+			{
+				isUpdating = true;
+				StartCoroutine(WaitForClient());
+			}
+		}
+
+		private IEnumerator WaitForClient()
+		{
+			yield return new WaitForSeconds(0.2f);
+			UpdateGUI();
+			isUpdating = false;
+		}
+
+		public void UpdateGUI()
+		{
+			UpdateResearchTechList();
+			UpdateFutureTechList();
+			UpdateAvailiableTechList();
+			PointLabel.MasterSetValue($"Available Points: {techWeb.researchPoints} (+1 / minute)");
```

Comment:
the (+1 / minute) is never going to change? is it hardcoded somewhere that the rate will always be 1+ per minute? its better to reference the value of how many points you'll receive in a given time incase it gets changed or becomes dynamic in the future.

Reply:
yeah fair enough